### ISSUE TITLE
Understand how _ignore_ attributes affects Enum subclasses

### ISSUE THREAD

**Issue by kstauffer** [python/mypy#12157]
**Feature**
In code like this, mypy should raise the indicated error:
```py
class Some(Enum):
   a = 1
   b = 2
   _ignore_ = ('a',)

reveal_type(Some.a) # E: "Type[Some]" has no attribute "a"
```

mypy should understand the effect that `_ignore_` has in Enum subclasses.

This issue is a follow-on to PR #12128.
